api/v1/User.php

```diff
@@ -367,4 +404,38 @@ public function eraseCredentials(): void
     {
         // irrelevant, but required function
     }
+
+    private function logUserLogOnOrOff(
+        bool $logOn,
+        int $userId,
+        string $userName,
+        int $sessionId,
+        int $countryId
+    ) : void {
+        $analyticsLogger = null;
+        try {
+            $legacyHelper = SymfonyToLegacyHelper::getInstance();
+            $analyticsLogger = $legacyHelper->getAnalyticsLogger();
+
+            $serverManager = ServerManager::getInstance();
+            $serverManagerId = Uuid::fromString($serverManager->getServerUuid());
+
+            $analyticsMessage = new UserLogOnOffSessionMessage(
+                $logOn,
+                new DateTimeImmutable(),
+                $serverManagerId,
+                $userId,
+                $userName,
+                $sessionId,
+                $countryId
+            );
+            $legacyHelper->getAnalyticsMessageBus()->dispatch($analyticsMessage);
+        } catch (Exception $e) {
```

Comment:
i would move the statements that do not throw any exceptions out of the try...catch  ..